### PR TITLE
(MODULES-2548) Initial modulesync

### PR DESCRIPTION
**Pull request by cyberious**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,40 +1,111 @@
...
+  elsif place_or_version =~ /^file:\/\/(.*)/
+    ['>= 0', { :path => File.expand_path($1), :require => false }]
+  else
+    [place_or_version, { :require => false }]
+  end
+end
 
 group :development do
-  gem "travis"
-  gem "travis-lint"
```
Comment (ferventcoder) [inline Gemfile]: Travis-lint is old and deprecated, but are there issues with removing travis gem?

Comment (cyberious) [inline Gemfile]: No they were used as part of the community development lifecycle.  Left over cruft from the rewrite

Comment (ferventcoder) [inline Gemfile]: okay, didn't realize that any possible testing of the .travis.yml file was still there or removed (if it even existed).

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  - PUPPET_GEM_VERSION: ~> 3.0
+    RUBY_VER: 200-x64
+  - PUPPET_GEM_VERSION: ~> 4.0
+    RUBY_VER: 21
+  - PUPPET_GEM_VERSION: ~> 4.0
+    RUBY_VER: 21-x64
+  - PUPPET_GEM_VERSION: 3.7.0
+    RUBY_VER: 193
+  - PUPPET_GEM_VERSION: 3.7.0
+    RUBY_VER: 200-x64
```
Comment (ferventcoder) [inline appveyor.yml]: 3.7.0 is the earliest then?

**Comment by ferventcoder** [comment]
Where is the .sync.yml?

**Comment by ferventcoder** [comment]
You didn't have any changes surrounding .travis.yml, but I would see removing the allow failures for 2.1.5 and then removing the future parser testing since we have v4 available. Thoughts?

**Comment by ferventcoder** [comment]
@cyberious you may want to set the permissions here appropriately. Ping @haus @joshcooper - can you make sure the right folks have access so this can be merged?

**Comment by ferventcoder** [comment]
Can we also make sure that https://travis-ci.org/puppetlabs/puppetlabs-iis is only running for PRs and branches with a .travis.yml?

**Comment by cyberious** [comment]
@ferventcoder permissions updated to match wsus_client, also I agree to remove allow_failures  and move to regular passing

**Comment by ferventcoder** [comment]
Guessing we can handle those things in modulesync updates?

File: .travis.yml [hunk 3]
```diff
@@ -14,10 +14,6 @@ matrix:
     env: PUPPET_GEM_VERSION="~> 4.0"
   - rvm: 2.0.0
     env: PUPPET_GEM_VERSION="~> 3.0" FUTURE_PARSER="yes"
-  - rvm: 1.9.3
-    env: PUPPET_GEM_VERSION="3.7.0"
-  - rvm: 2.0.0
-    env: PUPPET_GEM_VERSION="3.7.0"
```
Comment (ferventcoder) [inline .travis.yml]: whoops....

Comment (cyberious) [inline .travis.yml]: No oops, just using exactly as we had it previously in that we test ~3.0
You look at the file we have the same as what our default is for the configs.

Comment (ferventcoder) [inline .travis.yml]: I think we want to run against whatever the oldest supported will be. I'm not sure what version that is yet.

**Pull request closed without merging**